src/pages/DatasetCatalog.js

```diff
@@ -27,6 +27,21 @@ const canApplyForDataset = (dataset) => {
   return dataset.active && !isNil(dataset.dacId);
 };
 
+const extractDatasetProp = (propertyName, dataset) => {
+  const property = find({ propertyName })(dataset.properties);
+  return property?.propertyValue;
+};
+
+const isVisible = (dataset) => {
+  const openAccess = extractDatasetProp('Open Access', dataset);
+  if(!isNil(openAccess)){
+    // if open Access is false, dac approval required
+    return openAccess ? dataset.study.publicVisibility : (dataset.dacApproval && dataset.study.publicVisibility);
```

Comment:
`study` can be null, do we need null-safe traversal here?

Reply:
My thinking was since the openAccess is a property on a study,  if `openAccess` is not null then the study must exist.

Reply:
I think that's true today, but the code doesn't enforce that currently and could be incorrect in the near future.

Reply:
Good point. Do you think that a good approach would be to just add `dataset.study?.publicVisibility` or do more thorough handling here?

Reply:
I think that's fine for this case.